### PR TITLE
Add validations

### PR DESCRIPTION
**Pull request by Aafke89**
I added validations to all the models
- a refugee can now only send a reaction on a post to a dutchie and the other way around
- You cannot send a message to yourself
- A person can only make a post with the same object once
- A person cannot send two reactions on the same post

Hurray. check ook die custom validation. leuk voor later.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Post < ActiveRecord::Base
 
-  has_many :received_requests, class_name: 'Request'
+  has_many :received_requests, class_name: 'Request', dependent: :destroy
   belongs_to :author,  class_name: 'User', foreign_key: :author_id
 
-end
-# test
+  validates :subject, :content, :author_id, presence: true
```
Comment (MerelSteenbrink) [inline app/models/post.rb]: validates :postal_horn: :post_office: :postal_code :city_sunrise: :city
so: to be clear: we should validate city and address presence is true, right?

File: app/models/post.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+  validates :subject, uniqueness: {scope: :author_id, message: "You've already made a post with a subject"}
```
Comment (MerelSteenbrink) [inline app/models/post.rb]: message: "You've already mad a post with THIS subject" ???

File: app/models/request.rb [hunk 3]
```diff
@@ -2,5 +2,21 @@ class Request < ActiveRecord::Base
 
   belongs_to :post
   belongs_to :messenger, class_name: 'User', foreign_key: :messenger_id
+
+  validates :message, :status, :messenger_id, :post_id, presence: true
+  validates :status, :inclusion => {in: ["pending", "accepted", "declined"], allow_nil: false}
```
Comment (MerelSteenbrink) [inline app/models/request.rb:7]: default pending??

File: app/models/post.rb (hunk 2 above)
Comment (Aafke89) [inline app/models/post.rb]: Done!

File: app/models/post.rb (hunk 1 above)
Comment (Aafke89) [inline app/models/post.rb]: Yes, working on this

**Comment by MerelSteenbrink** [comment]
:palm_tree:

**Pull request merged**